docs/topics/Rich_Presence.md

```diff
@@ -115,137 +93,123 @@ typedef struct DiscordRichPresence {
 
 ![](rp-profile-view.png)
 
-## Secrets, secrets are...fun!
+## Joining
 
-Security is of the utmost importance to us here at Discord, and we know it is for you, too. That's why we want to make sure that you properly understand `matchSecret`, `joinSecret`, and `spectateSecret` so that your game data is safe and secure over the wire.
+Relevant Callbacks:
 
-Secrets are obfuscated data of your choosing. They could be match ids, player ids, lobby ids, or your favorite color in hexadecimal over and over. You should send us data that someone else's game client would need to join or spectate their friend. If you can't or don't want to support those actions, you don't need to send us secrets. In fact, you could skip the rest of this section, but we all know that the real magic is _learning_.
+    - `joinGame()`
+    - `joinRequest()`
 
-Your secrets on Discord need to be well-kept. When someone wants to spectate a friend's game, for example, they hit the "Spectate" button in their friend's profile. Discord checks to make sure their permissions are in order, and then passes their client the `spectateSecret` so the callback can fire on their machine, and their game client can handle it.
+Relevant Payload Data:
 
-To keep security on the up and up, Discord requires that you properly hash/encode/encrypt/put-a-padlock-on-and-swallow-the-key-but-wait-then-how-would-you-open-it your secrets. There are more than a couple of ways of going about this, but here's a couple points to muse over:
+    - `partyId`
+    - `partySize`
+    - `partyMax`
+    - `joinSecret`
+    - `matchSecret`
 
-#### What does your game require to join a party, spectate a match, or other actions available through Rich Presence?
+When you send the relevant payload data in the `Discord_UpdatePresence()` call, your player can invite a Discord chat channel to play with them. This invite is tied to the player's party information; if their `partyId` changes, the invite will expire. If their `partySize` and `partyMax` changes, the invite will add, remove, and fill up slots dynamically.
 
-The secrets you send us can be whatever data you want, so make life easy for yourself! If you need a `matchId`, `playerId`, `lobbyPassword`, and `bloodType` (we hope not), hash all of those up as your secret. Then, when you reverse the hash on callback, you'll have everything you need to complete the action.
+Other Discord users can click "Join" on the invitation. Their game will then launch, and the `joinGame()` callback will fire in their client with the inviting player's `joinSecret`. The client should reverse hash or otherwise unencrypt this secret and match the players together.
 
-#### How good is your security?
+To enable the Ask to Join button on your players' profiles, you'll need to be whitelisted by us—send an email to [gamedevs@discordapp.com](mailto:gamedevs@discordapp.com) to request access. Once whitelisted, you'll have access to the `joinRequest()` callback which sends the following data:
```

Comment:
this should be below the next heading

also pls make this an info/warn callout

Reply:
also they dont need a whitelist to get access to the callback.. they just need a whilelist for the button to even show/work in our client